### PR TITLE
feat: add tasks to use and install grype

### PR DESCRIPTION
**Pull request by osamamagdy**
Signed-off-by: osamamagdy <osamamagdy174@gmail.com>

### CODE REVIEW

File: tasks/supply-chain-security/task.yaml [hunk 1]
```diff
@@ -34,3 +34,19 @@ spec:
     echo $GIT_TOKEN | oras push -u $GIT_USER --password-stdin ghcr.io/$DOCKER_REGISTRY_ORG/$REPO_NAME:$VERSION-sbom /workspace/source/sbom.json
     echo $GIT_TOKEN | oras push -u $GIT_USER --password-stdin ghcr.io/$DOCKER_REGISTRY_ORG/$REPO_NAME:latest-sbom /workspace/source/sbom.json
 
+  - name: install-grype
+    image: alpine:3.16
+    resources: {}
+    script: |
+      #!/bin/bash
+      apk add --no-cache curl unzip
+      curl -sSfL https://raw.githubusercontent.com/anchore/grype/main/install.sh | \
```
Comment (ankitm123) [inline tasks/supply-chain-security/task.yaml:43]: Is there a way to verify that the binary is actually from grype? See codecov example: https://github.com/jenkins-x-plugins/jx-pipeline/blob/4950a36382b55f41a74f15b802e608340bfa4a9c/.lighthouse/jenkins-x/pullrequest.yaml#L35-L40

After all, this is all about supply chain!

File: tasks/supply-chain-security/task.yaml [hunk 2]
```diff
@@ -34,3 +34,19 @@ spec:
     echo $GIT_TOKEN | oras push -u $GIT_USER --password-stdin ghcr.io/$DOCKER_REGISTRY_ORG/$REPO_NAME:$VERSION-sbom /workspace/source/sbom.json
     echo $GIT_TOKEN | oras push -u $GIT_USER --password-stdin ghcr.io/$DOCKER_REGISTRY_ORG/$REPO_NAME:latest-sbom /workspace/source/sbom.json
 
+  - name: install-grype
+    image: alpine:3.16
+    resources: {}
+    script: |
+      #!/bin/bash
+      apk add --no-cache curl unzip
```
Comment (ankitm123) [inline tasks/supply-chain-security/task.yaml:42]: remove unzip

Comment (osamamagdy) [reply tasks/supply-chain-security/task.yaml]: done

File: tasks/supply-chain-security/task.yaml (hunk 1 above)
Comment (osamamagdy) [reply tasks/supply-chain-security/task.yaml:43]: This is handled inside the installation script itself. 
Refer to the main function and it uses download_and_install_asset at https://github.com/anchore/grype/blob/98104952120d3e368d265566b3d6b415e9da538a/install.sh#L681
And if we continue following the structure of all functions used we reach the hash_sha256_verify defined at
https://github.com/anchore/grype/blob/98104952120d3e368d265566b3d6b415e9da538a/install.sh#L272
So, they are actually doing verification steps internally

File: tasks/supply-chain-security/task.yaml [hunk 3]
```diff
@@ -34,3 +34,19 @@ spec:
...
+      #!/bin/bash
+      apk add --no-cache curl
+      curl -sSfL https://raw.githubusercontent.com/anchore/grype/main/install.sh | \
+      sh -s -- -b /workspace/source/bin
+      chmod +x /workspace/source/bin/grype
+  - name: scan-sbom
+    image:  alpine:3.16
+    resources: {}
+    script: |
+    #!/bin/bash
```
Comment (ankitm123) [inline tasks/supply-chain-security/task.yaml:50]: use sh instead of bash

**Pull request closed without merging**